azurerm/internal/services/storage/storage_object_replication_policy_resource.go

```diff
@@ -0,0 +1,362 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/storage/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceStorageObjectReplicationPolicy() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceStorageObjectReplicationPolicyCreate,
+		Read:   resourceStorageObjectReplicationPolicyRead,
+		Update: resourceStorageObjectReplicationPolicyUpdate,
+		Delete: resourceStorageObjectReplicationPolicyDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ObjectReplicationPolicyID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"source_storage_account_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.StorageAccountID,
+			},
+
+			"destination_storage_account_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.StorageAccountID,
+			},
+
+			"rules": {
+				Type:     pluginsdk.TypeSet,
+				Required: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"source_container_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.StorageContainerName,
+						},
+
+						"destination_container_name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validate.StorageContainerName,
+						},
+
+						// Possible values are "" means "OnlyNewObjects", "1601-01-01T00:00:00Z" means "Everything" and timeStamp "2020-10-21T16:00:00Z"
+						"copy_over_from_time": {
+							Type:         pluginsdk.TypeString,
+							Optional:     true,
+							Default:      "OnlyNewObjects",
+							ValidateFunc: validate.ObjectReplicationPolicyCopyOverFromTime,
+						},
+
+						"filter_prefix_matches": {
+							Type:     pluginsdk.TypeSet,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeString,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+						},
+
+						"rule_name": {
```

Comment:
i think we can remove rule as its in a block?
```suggestion
						"name": {
```